bundles/automation/org.eclipse.smarthome.automation.api/src/main/java/org/eclipse/smarthome/automation/RuleRegistry.java

```diff
@@ -44,10 +46,19 @@
  * </ul>
  *
  * @author Yordan Mihaylov - Initial Contribution
+ * @author Victor Toni - added search capability via {@link Predicate}
  */
 public interface RuleRegistry extends Registry<Rule, String> {
 
     /**
+     * This method is used to get a collection of {@link Rule}s which have the same {@link Predicate}.
+     *
+     * @param predicate to check the rules against
+     * @return collection of {@link Rule}s having matching predicate.
+     */
+    public Collection<Rule> getAll(Predicate<Rule> predicate);
```

Comment:
We must not create a dependency on Guava within our own API - this is very ugly.
We should also try to define a common scheme for doing filtered requests across our Registries (Items, Things, Rules, etc.). Currently, we have var args in place on the ItemRegistry and we had already discussed to introduce a common "Filter" mechanism.